packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -127,14 +127,19 @@ export const documentIdentifier = (result: Result): DocumentIdentifier => {
   };
 };
 
+const nonEmptyString = new StringValue({
+  required: false,
+  emptyAllowed: false,
+});
+
 const requiredNonEmptyString = new StringValue({
   required: true,
   emptyAllowed: false,
 });
 
 const rawPartialDefinition = {
   collection: requiredNonEmptyString,
-  author: requiredNonEmptyString,
+  author: nonEmptyString,
```

Comment:
Could the author be an empty string if the field was set to such during indexing?
Also, will headless/analytics crash if the author **is** an empty string? If not, I would just check that it is a string and log it.

```
author: new String()
```

Same comment for any other properties in the rawPartial definition.

More generally, I think it would be helpful if we knew what fields are always returned by the index, versus what fields are custom. We would then be able to reflect that in the `Result` and `Raw` interfaces,